src/units.cpp

```diff
@@ -25,4 +25,24 @@ void mass::serialize( JsonOut &jsout ) const
         jsout.write( string_format( "%d mg", value_ ) );
     }
 }
+
+template<>
+void length::serialize( JsonOut &jsout ) const
+{
+    if( value_ % 1'000'000 ) {
+        jsout.write( string_format( "%d km", value_ / 1'000'000 ) );
+    } else if( value_ % 1'000 ) {
+        jsout.write( string_format( "%d meter", value_ / 1'000'000 ) );
```

Comment:
Shouldn't this be 1000?
And all `value_ % 1xxx` checks be `value_ % 1xxx == 0` to avoid formatting `1` as `0 km` and `1000000` as `1000000 mm`